### PR TITLE
concavehexagon, tag, bottomroundrectangle shape implementations

### PR DESCRIPTION
**Pull request by d2fong**
Adds three new shapes
- bottomroundrectangle
- tag
- concave hexagon

### CODE REVIEW

**Comment by maxkfranz** [comment]
@d2fong Do you have pictures of each node shape so I can have a sense of what the code intends to do?

**Comment by d2fong** [comment]
screen shot 2017-06-06 at 1 36 09 pm

Here is a pic of the three shapes

**Comment by d2fong** [comment]
Left:  bottomroundrectangle Middle: concavehexagon Right: tag

**Comment by d2fong** [comment]
This is a continuation of the work to port some of the polygon shapes from the sbgn lib to cy.js.

concavehexagon and tag use the generatePolygon function.  bottomroundrectangle uses a similar approach for checkpoint and line intersection like the roundrectangle shape.

File: src/extensions/renderer/base/node-shapes.js [hunk 1]
```diff
@@ -348,6 +348,104 @@ BRp.generateBarrel = function(){
...
+        x, y,
+        nodeX,
+        nodeY,
+        width, height,
+        padding )
+      ;
+    },
+
+    // Looks like the width passed into this function is actually the total width / 2
+    checkPoint: function(
```
Comment (maxkfranz) [inline src/extensions/renderer/base/node-shapes.js]: I think the full width is supposed to be passed here.  Could we debug this Wed?

File: src/extensions/renderer/base/node-shapes.js [hunk 2]
```diff
@@ -348,6 +348,104 @@ BRp.generateBarrel = function(){
...
+        centerX - outerWidth, centerY,
+        centerX + outerWidth, centerY,
+        centerX + outerWidth, centerY - outerHeight
+      ];
+      if( math.pointInsidePolygonPoints( x, y, points)){
+        return true;
+      }
+
+
+      var checkInEllipse = function( x, y, centerX, centerY, width, height, padding ){
```
Comment (maxkfranz) [inline src/extensions/renderer/base/node-shapes.js]: Is is possible to use the same function as the roundrect or could this function be pulled out to `math` and possibly have the roundrect function use `checkInEllipse`?

**Comment by d2fong** [comment]
Okay, we can review in person on Wednesday.

**Pull request merged**